tests/test_client.py

```diff
@@ -0,0 +1,184 @@
+    podman_ids = podman_ids.split("\n")
+
+    for id in podman_ids:
+        subprocess.check_call(['podman', 'rm', '-f', id])
+
+
+@pytest.mark.katello_agent
+def test_install_subscription_manager(client, smoker):
+    client.run('yum -y install subscription-manager')
+    sub_man = client.package('subscription-manager')
+    assert sub_man.is_installed
+
+    hammer = smoker.package('tfm-rubygem-hammer_cli_katello')
+    assert hammer.is_installed
+
+
+@pytest.mark.katello_agent
+def test_register_subscription_manager_with_username_and_password(client, entities, server_hostname):
+    client.run("yum erase -y 'katello-ca-consumer-*'")
+
+    client.run_expect([0], "rpm -Uvh http://%s/pub/katello-ca-consumer-latest.noarch.rpm" % server_hostname)
+    client.run_expect([0], "subscription-manager register --force --org=%s --username=admin --password=changeme --env=Library" % entities['organization_label'])
+
+
+@pytest.mark.katello_agent
+def test_register_subscription_manager_with_activation_key(client, entities):
+    client.run_expect([0], "subscription-manager register --force --org=%s --activationkey=\"%s\"" % (entities['organization_label'], entities['activation_key']))
+
+    consumed = client.check_output("subscription-manager list --consumed")
+    assert entities['product'] in consumed
+
+
+@pytest.mark.katello_agent
+def test_host_is_registered(client, smoker):
+    hostname = client.check_output('hostname -s')
+    smoker.run_expect([0], "hammer host info --name %s" % hostname)
+
+
+@pytest.mark.katello_agent
+def test_enable_content_view_repo(client, entities):
+    enable = client.check_output("subscription-manager repos --enable=%s_%s_%s" % (entities['organization_label'], entities['product_label'], entities['yum_repository_label']))
+    assert "is enabled for this system" in enable
+
+
+def test_install_katello_host_tools(client, foreman_version):
+    release = client.system_info.release.split('.')[0]
+    client.run_expect([0], "yum -y install http://yum.theforeman.org/client/%s/el%s/x86_64/foreman-client-release.rpm" % (foreman_version, release))
+    client.run_expect([0], 'yum -y install katello-host-tools')
+    katello_host_tools = client.package('katello-host-tools')
+    assert katello_host_tools.is_installed
+
+
+def test_install_package_local(client):
+    client.run('yum -y install walrus-0.71')
+    walrus = client.package('walrus')
+    assert walrus.is_installed
+    assert walrus.version == '0.71'
+
+
+def test_available_errata(client, smoker, entities):
+    time.sleep(10)
+    hostname = client.check_output('hostname -s')
+    errata = smoker.check_output("hammer host errata list --host %s" % hostname)
+    assert entities['errata_id'] in errata
+
+
+@pytest.mark.katello_agent
+def test_install_katello_agent(client):
+    release = client.system_info.release.split('.')[0]
+
+    if release != 6 or release != 7:
+        pytest.skip("katello-agent not available on EL8")
```

Comment:
There are better ways to do this, and a good candidate for the server to inform through the API what is present.

Reply:
I think a fixture would work:
```python
@pytest.fixture
def katello_agent_client(client):
    release = client.system_info.release.split('.')[0]
    if release not in (6, 7):
        pytest.skip(f"katello-agent not available on EL{release}")
    return client
```

Then in all katello agent tests you would use this fixture.

Reply:
Or with the new `os_release` fixture:

```python
@pytest.fixture
def katello_agent_client(katello_client, os_release):
    if os_release not in ("6", "7"):
        pytest.skip(f"katello-agent not available on EL{release}")
    return katello_client
```

(you can't do pytest.skipIf on fixtures :()

Reply:
And now I wonder if this particular test is even really meaningful or if the fixture should do `yum install k-agent` and then  the other `test_katello_agent_*` tests can rely on that?